.github/workflows/ci.yml

```diff
@@ -54,15 +55,15 @@ jobs:
       matrix:
         include:
           - # If upgrading the Haskell image, also upgrade it in the lint job below
-            os: ["ubuntu-latest"]
-            image: haskell:9.2.8@sha256:b3b2f3909c7381bb96b8f18766f9407a3d6f61e0f07ea95e812583ac4f442cbb
-          - os: ["macOS-14"]
-          - os: ["windows-2019"]
-          - os: ["self-hosted", "macos", "ARM64"]
-          - os: ["self-hosted", "Linux", "ARM64"]
+            os: [ubuntu-24.04]
+            image: haskell:9.6.6
+          - os: [macos-15]
```

Comment:
Do we need to use macos 15? We should try to stay on older versions if possible, for library compatibility.

According to [GitHub docs](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) we should use both `macos-13` for x86, and `macos-14` for ARM builds.